ft/cachetable/cachetable.h

```diff
@@ -108,7 +108,14 @@ uint32_t toku_get_cleaner_iterations_unlocked (CACHETABLE ct);
 // create and initialize a cache table
 // size_limit is the upper limit on the size of the size of the values in the table
 // pass 0 if you want the default
-int toku_cachetable_create(CACHETABLE *result, long size_limit, LSN initial_lsn, struct tokulogger *logger);
+int toku_cachetable_create2(CACHETABLE *result, long size_limit,
+                           unsigned long client_pool_threads,
+                           unsigned long cachetable_pool_threads,
+                           unsigned long checkpoint_pool_threads,
+                           LSN initial_lsn, struct tokulogger *logger);
+
+#define toku_cachetable_create(r, s, l, o) \
+    toku_cachetable_create2(r, s, 0, 0, 0, l, 0);
```

Comment:
Instead of adding a MACRO, and using a less-than-optimal name with a '2' appended, couldn't we just put these arguments at the end of the function declaration and give them default values?

The other option is to create a second function with a different name that also creates a cacheable but with the optional manual thread count parameters.  Something like: toku_create_cachetable_with_thread_counts().  Then people would know there are two different behaviors and be forced to make an explicit choice while coding.